test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -467,14 +503,22 @@ void UcxContext::destroy_connections()
     }
 
     while (!_conns_in_progress.empty()) {
-        delete _conns_in_progress.begin()->second;
+        UcxConnection *conn = _conns_in_progress.begin()->second;
         _conns_in_progress.erase(_conns_in_progress.begin());
+        new UcxDisconnector(*conn);
     }
 
     UCX_LOG << "destroy_connections";
-    while (!_conns.empty()) {
-        // UcxConnection destructor removes itself from _conns map
-        delete _conns.begin()->second;
+    conn_map_t conns_tmp(_conns);
+    while (!conns_tmp.empty()) {
+        UcxConnection *conn = conns_tmp.begin()->second;
+        conns_tmp.erase(conns_tmp.begin());
+        new UcxDisconnector(*conn);
+    }
+
+    for (; !_conns.empty() && !_disconnecting_conns.empty();
+         ucp_worker_progress(_worker)) {
```

Comment:
pls use while loop

Reply:
ok, intention was do not call `ucp_worker_progress` extra time and avoid extra check in while loop